### PR TITLE
Add dialog to show all preferences in their raw form plus some filtering

### PR DESCRIPTION
**Pull request by simonharrer**
Relates to #859.

- Adds a dialog which shows all preferences in a table by their type, key, value and default value. 
- Table can be sorted
- Table can be filtered to show only the preferences which deviate from their default value. This is the most helpful one - and it revealed various bugs when having a closer look. :)
- Count is shown as well
- Reachable through Options -> Preferences -> Show Preferences (button is below import/export preferences)
- Converts default values of null to empty string

- [x] Change in CHANGELOG.md described?
- [x] Changes in pull request outlined? (What, why, ...)
- [ ] Tests created for changes? Mostly GUI changes, testing makes not that much sense here
- [x] Tests green?

### CODE REVIEW

**Comment by lenhard** [comment]
Looks good to me! Just rebase on master and this can be merged.

File: src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        table = new JTable();
+        table.setAutoCreateRowSorter(true);
+        updateModel();
+        panel.add(new JScrollPane(table), BorderLayout.CENTER);
+
+        this.getContentPane().add(panel);
+    }
+
+    private void updateModel() {
+        java.util.List<JabRefPreferencesFilter.PreferenceOption> preferenceOptions;
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java]: No need for fully qualified path?

File: src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+        private final List<JabRefPreferencesFilter.PreferenceOption> preferences;
+
+        public PreferencesTableModel(List<JabRefPreferencesFilter.PreferenceOption> preferences) {
+            this.preferences = Objects.requireNonNull(preferences);
+        }
+
+        @Override
+        public String getColumnName(int column) {
+            if (column == 0) {
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java]: switch?

File: src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        }
+
+        @Override
+        public Object getValueAt(int rowIndex, int columnIndex) {
+            if (rowIndex < 0 || rowIndex - 1 > preferences.size()) {
+                return "n/a";
+            }
+
+            JabRefPreferencesFilter.PreferenceOption preferenceOption = preferences.get(rowIndex);
+            if (columnIndex == 0) {
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java]: switch?

File: src/main/java/net/sf/jabref/JabRefPreferences.java [hunk 4]
```diff
@@ -543,8 +550,8 @@ private JabRefPreferences() {
         defaults.put(ENTRY_TYPE_FORM_WIDTH, 1);
         defaults.put(BACKUP, Boolean.TRUE);
         defaults.put(OPEN_LAST_EDITED, Boolean.TRUE);
-        defaults.put(LAST_EDITED, null);
-        defaults.put(LAST_FOCUSED, null);
+        defaults.put(LAST_EDITED, "");
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/JabRefPreferences.java]: Added twice (subject to only seeing fractions of the code...).

**Comment by oscargus** [comment]
Looks good! Some minor comments, which probably can be ignored/fixed later.

File: src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java (hunk 1 above)
Comment (simonharrer) [inline src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java]: Good point. Will fix this.

File: src/main/java/net/sf/jabref/JabRefPreferences.java (hunk 4 above)
Comment (simonharrer) [inline src/main/java/net/sf/jabref/JabRefPreferences.java]: Will have a look, thanks.

File: src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java (hunk 2 above)
Comment (simonharrer) [inline src/main/java/net/sf/jabref/JabRefPreferencesFilterDialog.java]: Na, I don't like switch statements, as one can miss the break.

**Pull request merged**